platforms/software/dependency-management/src/main/java/org/gradle/api/internal/attributes/FreezableAttributeContainer.java

```diff
@@ -55,12 +67,20 @@ public Set<Attribute<?>> keySet() {
 
     @Override
     public <T> AttributeContainer attribute(Attribute<T> key, T value) {
-        throw new IllegalArgumentException(String.format("Cannot change attributes of dependency %s after it has been resolved", owner.getDisplayName()));
+        if (delegate instanceof ImmutableAttributes) {
+            throw new IllegalArgumentException(String.format("Cannot change attributes of dependency %s after it has been resolved", owner.getDisplayName()));
```

Comment:
🤔 Does this class need a public `isFrozen()` method?  Also the error checking here is an exact duplicate that could be extracted.

Reply:
I don't think it needs a public `isFrozen` since at this point we never have the need for it. If we ever decide that we need to query the attribute container if its frozen, we could always add the method.

Reply:
I've extracted the common part.

Reply:
I'll Let It Go then. ❄️